confgenerator/confgenerator_test.go

```diff
@@ -102,40 +139,43 @@ func TestGenerateConfsWithValidInput(t *testing.T) {
 			}
 
 			// Retrieve the expected golden conf files.
-			expectedMainConfig := readFileContent(testName, goldenMainPath, t, true)
-			expectedParserConfig := readFileContent(testName, goldenParserPath, t, true)
+			expectedMainConfig := readFileContent(t, testName, platform.OS, goldenMainPath, true)
+			expectedParserConfig := readFileContent(t, testName, platform.OS, goldenParserPath, true)
+
 			// Generate the actual conf files.
-			mainConf, parserConf, err := uc.GenerateFluentBitConfigs(logsDir, stateDir, hostInfo)
+			mainConf, parserConf, err := uc.GenerateFluentBitConfigs(platform.defaultLogsDir, platform.defaultStateDir, platform.InfoStat)
 			if err != nil {
 				t.Fatalf("GenerateFluentBitConfigs got %v", err)
 			}
 			// Compare the expected and actual and error out in case of diff.
-			updateOrCompareGolden(t, testName, expectedMainConfig, mainConf, goldenMainPath)
-			updateOrCompareGolden(t, testName, expectedParserConfig, parserConf, goldenParserPath)
+			updateOrCompareGolden(t, testName, platform.OS, expectedMainConfig, mainConf, goldenMainPath)
+			updateOrCompareGolden(t, testName, platform.OS, expectedParserConfig, parserConf, goldenParserPath)
 
-			if platform == "windows" {
-				expectedOtelConfig := readFileContent(testName, goldenOtelPath, t, true)
-				otelConf, err := uc.GenerateOtelConfig(hostInfo)
+			if platform.OS == "windows" {
+				expectedOtelConfig := readFileContent(t, testName, platform.OS, goldenOtelPath, true)
+				otelConf, err := uc.GenerateOtelConfig(platform.InfoStat)
 				if err != nil {
-					t.Fatalf("GenerateOtelConfig got %v", err)
```

Comment:
The reasoning for keeping `Fatalf` applies here as well, doesn't it? Also in the non-Windows branch. You could then move the `updateOrCompareGolden` calls out of the `else`…